bin/release

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env ruby
+require 'pathname'
+require 'active_support/all'
+
+# path to your application root.
+APP_ROOT = Pathname.new File.expand_path("../../",  __FILE__)
+
+def run(command)
+  abort "command failed (#{$?}): #{command}" unless system command
+end
+
+Dir.chdir APP_ROOT do
+  puts %q[
+   _             _
+  | |           (_)
+  | | ___   __ _ _ _ __    __ _  _____   __
+  | |/ _ \ / _` | | '_ \  / _` |/ _ \ \ / /
+  | | (_) | (_| | | | | || (_| | (_) \ V /
+  |_|\___/ \__, |_|_| |_(_)__, |\___/ \_/
+            __/ |          __/ |
+           |___/          |___/
+  ]
+
+  RELEASE_DATE = 3.days.from_now.strftime('%Y-%m-%d')
+  RC_BRANCH = "stages/rc-#{RELEASE_DATE}"
+
+  DRY_RUN = %w[--dry-run -n].include?(ARGV.first)
+  ARGV.clear
+
+  puts "==> This script will perform all of the following:"
+  puts "1. Checkout a new RC branch based on stages/int"
+  puts "2. Merge master into the RC branch"
+  puts "3. Push the RC branch to stages/int on GitHub"
+  puts "4. Publish the wiki page for the RC"
+
+  print "Are you sure you want to proceed? (Type 'yes' to proceed, or anything else to abort) "
+  answer = gets
+  abort "==> You have chosen to abort the script." unless answer == "yes\n"
+
+  puts "==> Checking out master and pulling latest changes"
+  run "git checkout master && git pull" unless DRY_RUN == true
+
+  if `git branch`.include?('stages/int')
+    puts "==> Checking out stages/int and pulling latest changes"
+    run "git checkout stages/int && git pull" unless DRY_RUN == true
+  else
+    puts "==> Checking out new stages/int branch and pulling down origin/stages/int"
+    run "git checkout -b stages/int origin/stages/int" unless DRY_RUN == true
+  end
+
+  unless `git branch`.include?(RC_BRANCH)
+    puts "==> Checking out RC branch for release to prod 3 days from today"
+    run "git checkout -b #{RC_BRANCH}" unless DRY_RUN == true
+
+    puts "==> Merging master into RC branch"
+    run "git merge master" unless DRY_RUN == true
+
+    puts "==> Pushing RC branch to GitHub"
+    run "git push origin #{RC_BRANCH}" unless DRY_RUN == true
+
+    # puts "===> Merge RC branch into origin/stages/int"
+    # run "git push origin #{RC_BRANCH}:stages/int" unless DRY_RUN == true
+  end
+end
+
+Dir.chdir ENV['HOME'] do
+  if `ls -d */`.include?('login-dot-gov')
```

Comment:
This make some assumptions about how I have these repos saved on my machine. This will likely be pretty different for different developers who have their own preferences for how they organize things on their machine.

For example, all of my repos are in an `identity` folder and the names are the repo name on GH with the `identity-` lopped off, e.g. `identity/idp`, `identity/devops` etc. James has all of his in a `Code` directory with the original repo name as the filename.

Reply:
The only assumption this is making is that you don't already have a directory in your home folder called `login-dot-gov`. It is creating a new directory called `login-dot-gov` within your home directory, which I thought would be an unlikely name that anyone is using. Inside this directory, it clones the devops and Wiki repos (without affecting the directories you use on a regular basis).

Reply:
Ah, I thought this was trying to update the repos I use for my work which would be frustrating.

Only thought from here is that is would be nice to use something like `/tmp/login-dot-gov` or `~/.tmp/login-dot-gov` instead, and also could it clean up after itself when it is done?